conans/test/integration/toolchains/cmake/test_cmaketoolchain.py

```diff
@@ -354,3 +355,48 @@ def layout(self):
 
     presets = load_cmake_presets(os.path.join(build_dir, "conan"))
     assert presets["configurePresets"][0]["binaryDir"] == build_dir
+
+
+def test_cmake_presets_multiconfig():
+    client = TestClient()
+    profile = textwrap.dedent("""
+        [settings]
+        os = Windows
+        arch = x86_64
+        compiler=msvc
+        compiler.version=193
+        compiler.runtime=static
+        compiler.runtime_type=Release
+    """)
+    client.save({"conanfile.py": GenConanfile(), "profile": profile})
+    client.run("create . mylib/1.0@ -s build_type=Release --profile:h=profile")
+    client.run("create . mylib/1.0@ -s build_type=Debug --profile:h=profile")
+
+    client.run("install mylib/1.0@ -g CMakeToolchain -s build_type=Release --profile:h=profile")
+    presets = json.loads(client.load("CMakePresets.json"))
+    assert len(presets["buildPresets"]) == 1
+    assert presets["buildPresets"][0]["configuration"] == "Release"
+
+    client.run("install mylib/1.0@ -g CMakeToolchain -s build_type=Debug --profile:h=profile")
+    presets = json.loads(client.load("CMakePresets.json"))
+    assert len(presets["buildPresets"]) == 2
+    assert presets["buildPresets"][0]["configuration"] == "Release"
+    assert presets["buildPresets"][1]["configuration"] == "Debug"
+
+    client.run("install mylib/1.0@ -g CMakeToolchain -s build_type=RelWithDebInfo --profile:h=profile")
+    client.run("install mylib/1.0@ -g CMakeToolchain -s build_type=MinSizeRel --profile:h=profile")
+    presets = json.loads(client.load("CMakePresets.json"))
+    assert len(presets["buildPresets"]) == 4
+    assert presets["buildPresets"][0]["configuration"] == "Release"
+    assert presets["buildPresets"][1]["configuration"] == "Debug"
+    assert presets["buildPresets"][2]["configuration"] == "RelWithDebInfo"
+    assert presets["buildPresets"][3]["configuration"] == "MinSizeRel"
+
+    # Repeat one
+    client.run("install mylib/1.0@ -g CMakeToolchain -s build_type=Debug --profile:h=profile")
+    client.run("install mylib/1.0@ -g CMakeToolchain -s build_type=Debug --profile:h=profile")
+    assert len(presets["buildPresets"]) == 4
+    assert presets["buildPresets"][0]["configuration"] == "Release"
+    assert presets["buildPresets"][1]["configuration"] == "Debug"
+    assert presets["buildPresets"][2]["configuration"] == "RelWithDebInfo"
+    assert presets["buildPresets"][3]["configuration"] == "MinSizeRel"
```

Comment:
I don't know if it's needed but should we run a `client.run("create . mylib/1.0@ -s build_type=RelWithDebInfo --profile:h=profile")`, for example, and check that it's running ok?

Reply:
Well, this test is an empty conanfile that can run in any system, and actually, the `buildPresets` are not parsed/used by the cmake build helper, they are only for the consumers using the `CMakePresets.json`

Reply:
Got it